client/src/components/QualityManagement/Parts.jsx

```diff
@@ -0,0 +1,328 @@
+      newSelected = newSelected.concat(selected.slice(0, -1));
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(
+        selected.slice(0, selectedIndex),
+        selected.slice(selectedIndex + 1)
+      );
+    }
+
+    setSelected(newSelected);
+  };
+
+  const handleDelete = async (e) => {
+    e.preventDefault();
+    try {
+      await deleteDefects(selected);
+      toast({
+        title: 'Deleted',
+        description: 'Defects have been deleted successfully',
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+      setSelected([]);
+      refetch();
+    } catch (err) {
+      toast({
+        title: 'Error',
+        description: err.message,
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setIsLoadingButton(true);
+    try {
+      await createDefect({
+        partName: partName,
+        type: type.current.value,
+        status: status.current.value,
+        description: description.current.value,
+        onCloseComplete: onClose,
+      });
+      toast({
+        title: 'Request Sent',
+        description: 'Defect has been created successfully',
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+      refetch();
+    } catch (err) {
+      toast({
+        title: 'Error',
+        description: err.message,
+        status: 'error',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+    setIsLoadingButton(false);
+  };
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  const addDefectForm = () => (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Add a new Defect</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody pb={6}>
+          <FormControl isRequired>
+            <FormLabel>Part Name</FormLabel>
+            <Input onChange={handlePartNameInput} value={partName} placeholder="Part name" />
+          </FormControl>
+
+          <FormControl mt={4}>
+            <FormLabel>Defect Type</FormLabel>
+            <Select ref={type}>
+              <option>Broken</option>
+              <option>Reparable</option>
+            </Select>
+          </FormControl>
+          <FormControl mt={4}>
+            <FormLabel>Defect Status</FormLabel>
+            <Select ref={status}>
+              <option>Solved</option>
+              <option>Pending</option>
+              <option>Ongoing</option>
+            </Select>
+          </FormControl>
+          <FormControl mt={4}>
+            <FormLabel>Description</FormLabel>
+            <Textarea ref={description} placeholder="Description" />
+          </FormControl>
+        </ModalBody>
+
+        <ModalFooter>
+          <Button
+            isLoading={isLoadingButton}
+            disabled={partName ? false : true}
```

Comment:
```suggestion
            disabled={!partName}
```
Would give the same effect I think